README

			----------------------------------------------------
			       ------------ proxybot 0.3 ------------
			----------------------------------------------------
			    --------------- by: hysterix ---------------

Requirements:
 apache, php 5, curl


What:
	Feature List:
		- scour sites for proxies
		- multi-threaded proxy testing
		- multi-threaded proxy anonymity checking (internal or external check)
		- output proxies in certain formats
		- banned list
		- scour sites for new codeen/planet proxies list
		- db maintenance section

		
How:
	Select your mode:
	- search
	- test
	- output
	- banned
	- database
	- help

	search - select which sites you want to scour for proxies and configure the output once the list is complete
		 - you can save the list into the database to be checked later
		 - you can output the list as plaintext or csv
		 - you can choose which sites you want to search for proxies
		 - select mysql as the option in order to check proxy status

	test   - select which proxies you want to test
		 - you can test new proxies entered from the previous search
		 - you can test good proxies to make sure they are still good
		 - you can test inactive proxies to see if they might be up
		 - you can test the anonymity level of proxies after they are known to be good
		 - if a new proxy scoured never works once, it is deleted from the database
		 - if a good or anonymous proxy is down, it is put into the inactive table
		 - if an inactive proxy is tested and is still down, it stays in the inactive table
		 - if an inactive proxy is tested and is up it is put into the good table
		 - there are over 9000 new, good, and inactive proxies allowed per check by default

	output - select how you want your final list format
		 - you can output anonymous proxies, good, new, inactive, or banned proxies
		 - you can output the proxy list as plaintext, csv, or as proxychains format: example - (socks4 192.168.1.49 1080)

	banned - scour sites for new codeen/planet proxies to ban
		 - you can save the output to mysql in the banned table
		 - you can output as a plaintext list
		 - you can choose which sites you want to search for new codeen/planet proxies
	
	database - perform simple maintenance on the database
		 - you can select both the new + good table, new, good, inactive, or banned table
		 - you can eliminate duplicates, or truncate table
	help   - very simple in program help menu
	
	setup  - there is no update from v0.2 -> v0.3. completely remove all tables and data as the table structure has changed
		   - delete entire proxybot directory and load new directory in its place; files have been changed, added, and removed
		   - all database info is inside of config.inc.php; edit this file depending on your database setup
		   - all other configuration for proxybot is in config.php; change depending on host settings
		   - set ALLOW_TRUNCATE to 1 in config.php if you want to allow deleting of tables; not a good idea on a live site
		   - free host disclaimer:
			  proxybot does not work on free hosts for the following reasons:
				1. most free hosts have a very low default timeout; the script needs time to scour and to check
				2. lowering the number of proxies to check does not really help on free hosts; the checking feature does not work at all
				3. CURLOPT_FOLLOWLOCATION + php + open_basedir or safe_mode = fuckup
					see: http://us.php.net/manual/ro/function.curl-setopt.php#71313 for workaround; don't bother, it still failed us
				4. free hosts most likely have a max number of curl requests and proxybot surely exceeds it
			
				
Why:
	Why did I include a gnu gpl?  As the level of sophistication of this program grew to rival that of commercial software out there, 
	I realized the necessity of a license; to prevent this code from ever being anything but 100% open source, and 100% public domain.

	quote of the day:
	 "As you think, so shall you become." - Bruce Lee


Credits:
 	curl class: David - semlabs.co.uk
	  new logo: dr_eval
inspiration to 
 cut down bulk: pedrobear

HISTORY
- 0.3 - 
	- default proxy timeout value raised from three to five seconds
	- default site timeout value raised from fifteen to 120 seconds
	- default value of max number of new and good proxies IS OVER 9000 (due to large number of proxies returned per click, and high speed of proxy checking)
	- default behavior not to display the number of new or good proxies that can be checked
	- default number of max simultaneous connections is 100
	- default choice for search tab changed to mysql
	- fixed dotted border around menu links css bug
	- fixed hidemyass md5 hashing bug; was hashing http header info corrupting hash value
	- fixed curl header caching bug
	- fixed 'go back' button bug on displaying pages
	- added curlopt_fresh_connect to prevent retrieving cached pages
	- added proxy new/good/anon/dead/banned display to upper right corner on all tabs
	- added loading screen for initial proxy scrape
	- added database maintenance tab, enabling duplicate removal and table truncate
	- added banned table/section, enabling easy codeen/planet proxy banning
	- added default planetlab/codeen banned list
	- added proxy anonymous level checker
	- added inactive table; now no good proxies are ever deleted, only stored for checking at a later time
	- added test dead proxies to testing tab
	- user can now modify max number of simultaneous curl connections (MAX_CONNECTIONS)
	- user can now turn on or off automatic duplicate removal (AUTO_DUPE_REMOVE) - default behavior is on
	- user can now display unknown hidemyass hashes/port images inside proxybot for stupidly easy hash hijacking - (HIDEMYASS_HASHIFIER) - default is off	
	- user can now change time-out value of site loading (SITE_TIMEOUT)
	- user can now change max number of new proxies to be checked per click (MAX_NEW_CHECK)
	- user can now change max number of good proxies to be checked per click (MAX_GOOD_CHECK)
	- user can now change whether or not the value of new proxies to be checked is displayed (SHOW_NEW_CHECK)
	- user can now change whether or not the value of good proxies to be checked is displayed (SHOW_GOOD_CHECK)
	- user can now change whether or not to delete banned ips before testing (AUTO_BAN) - default on
	- user can now turn on or off the truncate function (ALLOW_TRUNCATE) - default behavior is off
	- user can now display hide my ass port numbers and hashes from within proxybot
	- user can now change database table names in config.inc.php file ($config['tbl_new'], $config['tbl_good'], $config['tbl_inactive'], $config['tbl_banned'])
	- user can now select all/none for easy proxy scraping choosing 
	- user can now output anonymous proxies only if desired
	- modified charon codebase to enable multi-threaded proxy anonymous level checking, as well as added
	  the ability to check anonymity level of proxies without making requests back to self as charon originally did	
	- gui cleaned up for all tabs
	- csv.php, plaintext.php, and proxychains.php was depricated by output.php
	- split_url.php and simple_html_dom.php have both been depricated in favor of custom regex
	- number of proxies scraped as well as script execution time is now displayed
	- both grey and white hidemyass port images are grabbed per click if hashifier option is on, ensuring all hashes are recovered
	- fetching hidemyass proxies has been significantly sped up due to multi-threading, and other function optimizations
	- background image updated; now with more hofmann and mandelbrot (adding to the size)
	- logo image updated - thanks dr_eval
	- user-agent strings file reduced; only the most common user agents used - thanks pedrobear
	- included a gnu license to ensure proxybot remains 100% free and in the public domain continually
	
	-= new sites added =-
		- 2ch.net
		- atomintersoft.com
		- comp-info.ru
		- cybersyndrome.net
		- freeproxy.ch
		- freeproxy.ru
		- freeproxylists.com
		- getfreeproxy.info
		- ip-adress.com
		- ipcn.org
		- j1f.net
		- my-proxy.com
		- proxiesthatwork.com
		- proxy-heaven.blogspot.com
		- proxy-list.net
		- proxygo.com.ru
		- proxyleech.com
		- proxylists.net
		- proxyserverfinder.com
		- rosinstrument.com
		- speedtest.at 
		- xroxy.com
		
	-= old sites fixed =-
		- hidemyass.com - direct linking to /hide-planetlab/ no longer exists; changed url
		  and added a significant number of md5 hashes to the ghetto tables flat file
		- samair.ru obfuscated the previously scraped page with javascript;
		  changed to smaller, but bot friendly page
		  
	-= old sites removed =-
		- bestproxylist.cn: no such animal exists; gone, never to be heard from again
	
- 0.2 - 
	- fixed proxy test function to work correctly on more hosts
	- fixed bug in proxy retest function
	- lowered time-out value of proxy checking function, increasing speed
	- user can now change time-out value in config file (PROXY_TIMEOUT)
	- raised the number of good proxies to check at a time from 10 to 50


includes/config.php
<?php

/*
	config.php - config for proxybot
*/

session_start(); // start the session

include_once('proxybot.class.php');	// proxybot class
include_once('curl.php');    		// very nice curl wrapper, in fact, the only library worth keeping

//error_reporting(E_ALL);
//ini_set('display_errors','1');

set_time_limit(0);			    	// stop timeouts
ini_set('memory_limit','128M');		// ever since multi-threading through (whatever number you want) proxies at a time was implemented, moar mem is needed

/* connectivity */
define('MAX_CONNECTIONS',100); 		// max number of simultaneous curl connections (crank it up for serious speed (high memory usage) - keep it high to keep speed up)
define('MAX_RETRYS',2);	  	   		// max number of retrys per site
define('PROXY_TIMEOUT',5);	   		// timeout value to check a proxy in seconds
define('SITE_TIMEOUT',120);	   		// timeout value when connecting to sites

/* testing */
define('MAX_NEW_CHECK',9001);  		// max number of new proxies that could potentially be checked per user click
define('MAX_GOOD_CHECK',9001); 		// max number of good proxies that could potentially be checked per user click
define('MAX_INAC_CHECK',9001); 		// max number of inactive proxies that could potentially be checked per user click
define('PROXY_REQ_VAR','partyvan'); // change this to whatever you want

/* gui */
define('SHOW_NEW_CHECK',0);	  		// 0 - do not show the number of new proxies that can be checked;  1 - show
define('SHOW_GOOD_CHECK',0);  		// 0 - do not show the number of good proxies that can be checked; 1 - show

/* behavior */
define('AUTO_DUPE_REMOVE',1);		// 0 - do not automatically remove duplicates; 1 - automatically remove duplicates before testing any proxies, and after inserting new ones
define('AUTO_BAN',1);				// 0 - do not automatically delete banned ips; 1 - automatically delete banned ips
define('ALLOW_TRUNCATE',0);			// 0 - do not allow the truncate function; 1 - allow the truncate function
define('HIDEMYASS_HASHIFIER',0);	// 0 - do not display hash and portnumber of unknown hidemyass hashes; 1 - show  -- sometimes when turned on, it randomly spits out errors finding hashes even though we already have them; be prepared for this, couldnt figure out why it does it
define('HIDEMYASS_RAPE_LEVEL',15);	// how badly do you want to rape hidemyass.com? This is the amount of threaded hidemyass urls scraped per SINGLE user click (shouldnt go higher than around 20)

/* metadata */
define('VERSION',0.3);				// it seemed liked v0.1 was only yesterday...

$header = '<html><head><script language="javascript" src="includes/utils.js"></script>
		  <link rel="stylesheet" href="css/eat_tabs.css">
	          <link rel="stylesheet" href="css/yastyle.css"><title>proxybot v'.VERSION.'</title></head><body>'."\n\n";

$logo   = '<div class="main">
	    <div class="title"><div class="pvers">proxybot v'.VERSION.'</div>
	     <div class="logo"><div class="byme">by: hysterix</div><img src="images/logo-small.png" alt="skull" height="200" width="200" /></div>
	    </div>';


?>
